opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -424,4 +436,31 @@ mod tests {
             matches!(result.unwrap_err(), MetricsError::Other(err) if err == "reader is not registered")
         );
     }
+
+    fn collection_triggered_by_interval_helper<RT>(runtime: RT)
+    where
+        RT: crate::runtime::Runtime,
+    {
+        let interval = std::time::Duration::from_millis(1);
+        let exporter = InMemoryMetricsExporter::default();
+        let reader = PeriodicReader::builder(exporter.clone(), runtime)
```

Comment:
Do we expect to merge #2142 soon? If that's the case, then these tests would have to be removed/skipped. Maybe add these tests post #2142 once we have a mechanism to accept custom runtimes?

Reply:
I am using these tests to primarily validate that when 2142 approach is merged, we'll not have hangs/deadlocks, not matter which async runtime user is on.

(Ofcourse, will modify these tests to match 2142 later, as there will be no need of specifying runtime, and reducing test matrix too)

Reply:
Not as inconvenient but I guess the simpler thing (for review) would be to have the tests added in #2142 itself. Otherwise, we just increase the amount of diff for review.